neo4j-cli/aura/internal/subcommands/graphanalytics/session/list.go

```diff
@@ -0,0 +1,62 @@
+package session
+
+import (
+	"net/http"
+
+	"github.com/neo4j/cli/common/clicfg"
+	"github.com/neo4j/cli/neo4j-cli/aura/internal/api"
+	"github.com/neo4j/cli/neo4j-cli/aura/internal/output"
+	"github.com/spf13/cobra"
+)
+
+func NewListCmd(cfg *clicfg.Config) *cobra.Command {
+	var projectId string
+	var instanceId string
+	var organizationId string
+
+	cmd := &cobra.Command{
+		Use:   "list",
+		Short: "Returns a list of Graph Analytics Serverless sessions",
+		Long: `This subcommand returns a list containing a summary of each of your Graph Analytics Serverless session
+				By default, this subcommand lists all sessions a user has access to across all projects.
+				You can filter sessions in a particular project using:
+				--organization-id <organization-id>
+				--project-id <project-id>
+				--instance-id <instance-id>
+`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			path := "/graph-analytics/sessions"
+
+			queryParams := make(map[string]string)
+			if organizationId != "" {
+				queryParams["organizationId"] = organizationId
+			}
+			if projectId != "" {
+				queryParams["projectId"] = projectId
+			}
+			if instanceId != "" {
+				queryParams["instanceId"] = instanceId
+			}
+
+			cmd.SilenceUsage = true
+			resBody, statusCode, err := api.MakeRequest(cfg, path, &api.RequestConfig{
+				Method:      http.MethodGet,
+				QueryParams: queryParams,
+			})
+			if err != nil {
+				return err
+			}
+
+			if statusCode == http.StatusOK {
+				output.PrintBody(cmd, cfg, resBody, []string{"id", "name", "status", "project_id", "cloud_provider", "ttl"})
+			}
+			return nil
+		},
+	}
+
+	cmd.Flags().StringVar(&projectId, "project-id", "", "An optional Project ID to filter sessions in a project")
+	cmd.Flags().StringVar(&organizationId, "organization-id", "", "An optional Organization ID to filter sessions in an organization")
```

Comment:
This change introduces massive inconsistencies into the CLI. The CLI has yet to be updated to use the new "project" nomenclature and still uses "tenant" for this purpose. I'm also massively unfamiliar with the new nomenclature, what is "organization" and the "old" equivalent?

These changes are being added to Version 1 of the API? When this Graph Analytics stuff is being added the API is going to be half using "tenant" and half using "project"? I think this should be revisited. Perhaps these changes should be waiting until Version 2 of the API.

Reply:
organization sits on top of multiple projects. see https://neo4j.com/docs/aura/visual-tour/#_console_structure

AFAIK, there is a plan to add more endpoints to the Aura-API which might include managing  organizations.

As its an optional filter we can also omit it for now.

About project vs tenant.
How about the user can still specify it as tenant-id then this discussion is not part of the Aura CLI and we only map it back to project-id inside the endpoint?

These endpoints are exposed as shown so we cannot break it anymore.
But we could also allow both project-id/tenant-id for consistency with other endpoints.

Previous discussion around the api are listed here in this [doc](https://docs.google.com/document/d/1bIliucbkon6GUP8WrQzpZ9Y3-t8l2GrX8XVcQn8lOFY/edit?tab=t.0)

Reply:
Out of interest, where are these endpoints exposed? There's nothing about graph analytics in the API spec: https://neo4j.com/docs/aura/platform/api/specification/?urls.primaryName=Aura%20v1

Reply:
waiting for a PR review to get these documented in the spec see https://github.com/neo4j/aura-cli/pull/116#issuecomment-2829977422

Reply:
I see I see, so these endpoints are available just not documented! Thanks!

I think I might stand by having `tenant-id` in the CLI for now - just for consistencies sake!

Reply:
Right, I'm beginning to understand. I also notice that the namespacing is inconsistent between the CLI and API. Here we have some `graph-analytics` namespace (which I must say, feels nice), whilst in the API we just jump directly into `sessions` without this prefix? Apologies, I realise that this isn't directly reviewing the changes in question here, I just want to make sure we're making the right design decisions.

Reply:
@FlorentinD I'd check in with the aura-console folks re: projects / tenants and the Aura API.  I had heard that projects would only appear in aura app v2 and the current aura api will stick with tenants

As fars as this version of the Aura CLI is concerned, we should be talking about tenants.   

When the Aura CLI gets updated for Aura API v2, then we'll switch to projects - everywhere in the CLI.

Reply:
> Right, I'm beginning to understand. I also notice that the namespacing is inconsistent between the CLI and API. Here we have some `graph-analytics` namespace (which I must say, feels nice), whilst in the API we just jump directly into `sessions` without this prefix? Apologies, I realise that this isn't directly reviewing the changes in question here, I just want to make sure we're making the right design decisions.

could call - i forgot to add the namespace on the other PR. Thanks for spotting